meson.build

```diff
@@ -0,0 +1,350 @@
+config_h.set_quoted('PACKAGE_URL', 'https://tukaani.org/xz/')
+
+# host
+
+config_h.set10('WORDS_BIGENDIAN', host_machine.endian() == 'big',
+    description: 'Define to 1 if the processor stores words with the most significant byte first.'
+)
+
+host_os = host_machine.system()
+
+cygwin = 'cygwin'
+windows = 'windows'
+linux = 'linux'
+sunos = 'sunos'
+asm_os = [ 'linux', 'dragonfly', 'freebsd', 'netbsd', 'openbsd', 'windows', 'cygwin' ]
+sys_cygwin = cygwin.contains(host_os)
+sys_windows = windows.contains(host_os)
+
+COND_W32 = sys_cygwin or sys_windows
+
+# binaries
+cc = meson.get_compiler('c')
+
+# try mimic AC_USE_SYSTEM_EXTENSIONS
+if host_machine.system() == 'linux'
+    add_project_arguments('-D_GNU_SOURCE', language: 'c')
+elif host_machine.system() == 'sunos'
+    add_project_arguments('-D__EXTENSIONS__', language: 'c')
+    add_project_arguments('-D_POSIX_PTHREAD_SEMANTICS', language: 'c')
+else
+    add_project_arguments('-D_TANDEM_SOURCE', language: 'c')
+    add_project_arguments('-D_ALL_SOURCE', language: 'c')
+endif
+
+lzma_c_args = []
+cflags_try = []
+
+if cc.get_argument_syntax() == 'gcc'
+    cflags_try += [
+        '-Wvla',
+        '-Wformat=2',
+        '-Winit-self',
+        '-Wmissing-include-dirs',
+        '-Wshift-overflow=2',
+        '-Wstrict-overflow=3',
+        '-Walloc-zero',
+        '-Wduplicated-cond',
+        '-Wfloat-equal',
+        '-Wundef',
+        '-Wshadow',
+        '-Wpointer-arith',
+        '-Wbad-function-cast',
+        '-Wwrite-strings',
+        '-Wdate-time',
+        '-Wsign-conversion',
+        '-Wfloat-conversion',
+        '-Wlogical-op',
+        '-Waggregate-return',
+        '-Wstrict-prototypes',
+        '-Wold-style-definition',
+        '-Wmissing-prototypes',
+        '-Wmissing-declarations',
+        '-Wredundant-decls',
+#
+        '-Wc99-compat',
+        '-Wc11-extensions',
+        '-Wc2x-compat',
+        '-Wc2x-extensions',
+        '-Wpre-c2x-compat',
+        '-Warray-bounds-pointer-arithmetic',
+        '-Wassign-enum',
+        '-Wconditional-uninitialized',
+        '-Wdocumentation',
+        '-Wduplicate-enum',
+        '-Wempty-translation-unit',
+        '-Wflexible-array-extensions',
+        '-Wmissing-variable-declarations',
+        '-Wnewline-eof',
+        '-Wshift-sign-overflow',
+        '-Wstring-conversion',
+    ]
+
+    lzma_c_args += cc.get_supported_arguments(cflags_try)
+endif
+
+config_h.set10('HAVE_VISIBILITY', cc.has_function_attribute('visibility'),
+    description: 'Define to 1 if the compiler supports simple visibility declarations.'
+)
+
+ctor_fct_src = '''
+__attribute__((__constructor__))
+static void my_constructor_func(void) { return; }
+int main() { return 0; }
+'''
+if cc.compiles(ctor_fct_src, name : 'constructor attribute for functions')
+    config_h.set10('HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR', true,
+        description: 'Define to 1 if __attribute__((__constructor__)) is supported for functions.'
+    )
+endif
```

Comment:
Simpler:

```
if cc.has_function_attribute('constructor')
    config_h.set('HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR', 1,
        description: 'Define to 1 if __attribute__((__constructor__)) is supported for functions.')
endif
```

Reply:
done in a83a638d7a129a69f54e5c9a0b86eee96dfaaa56